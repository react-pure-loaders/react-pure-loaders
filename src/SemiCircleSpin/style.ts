import { keyframes } from '@emotion/react';
import { globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const size = '35px';
const pos = 30;

const spinRotate = keyframes({
  '0%': {
    transform: 'rotate(0deg)',
  },
  '100%': {
    transform: 'rotate(360deg)',
  },
  '50%': {
    transform: 'rotate(180deg)',
  },
});

export default (color?: string) => ({
  height: size,
  overflow: 'hidden',
  position: 'relative' as 'relative',
  width: size,
  ' > div': {
    ...globalAnimation(),
    animation: `${spinRotate} 0.6s 0s infinite linear`,
    backgroundImage: `linear-gradient(transparent 0%, transparent ${(100 - pos)}%, ${color || PRIMARY_COLOR} ${pos}%, ${color || PRIMARY_COLOR} 100%)`,
    borderRadius: '100%',
    borderWidth: '0px',
    height: '100%',
    position: 'absolute' as 'absolute',
    width: '100%',
  },
});
