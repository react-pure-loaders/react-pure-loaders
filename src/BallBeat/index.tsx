/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
