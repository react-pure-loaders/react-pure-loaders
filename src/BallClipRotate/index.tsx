/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
