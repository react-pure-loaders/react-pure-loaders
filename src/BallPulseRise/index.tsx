/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
