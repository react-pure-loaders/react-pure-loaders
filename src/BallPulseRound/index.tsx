/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
