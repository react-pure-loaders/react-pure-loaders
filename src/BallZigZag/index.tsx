/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
