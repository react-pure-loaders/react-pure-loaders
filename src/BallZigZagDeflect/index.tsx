/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
