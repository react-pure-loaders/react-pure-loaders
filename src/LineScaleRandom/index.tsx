/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
