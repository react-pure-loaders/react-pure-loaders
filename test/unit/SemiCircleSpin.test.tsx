import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import SemiCircleSpin from '../../src/SemiCircleSpin';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<SemiCircleSpin>', () => {
    afterEach(cleanup);

    test('SemiCircleSpin should match snapshot', () => {
        const { container } = render(<SemiCircleSpin loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('SemiCircleSpin should have default color', () => {
        const { container } = render(<SemiCircleSpin loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'background-image',
            `linear-gradient(transparent 0%,transparent 70%,${PRIMARY_COLOR} 30%,${PRIMARY_COLOR} 100%)`,
            {
                target: '> div',
            },
        );
    });

    test('SemiCircleSpin should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<SemiCircleSpin color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule(
            'background-image',
            `linear-gradient(transparent 0%,transparent 70%,${color} 30%,${color} 100%)`,
            {
                target: '> div',
            },
        );
    });

    test('SemiCircleSpin should have no children', () => {
        const { container } = render(<SemiCircleSpin loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('SemiCircleSpin should have three children', () => {
        const { container } = render(<SemiCircleSpin loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(2);
    });
});
