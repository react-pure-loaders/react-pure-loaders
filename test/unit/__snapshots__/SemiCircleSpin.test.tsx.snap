// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SemiCircleSpin> SemiCircleSpin should match snapshot 1`] = `
@keyframes animation-0 {
  0% {
    -webkit-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -ms-transform: rotate(0deg);
    transform: rotate(0deg);
  }

  100% {
    -webkit-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -ms-transform: rotate(360deg);
    transform: rotate(360deg);
  }

  50% {
    -webkit-transform: rotate(180deg);
    -moz-transform: rotate(180deg);
    -ms-transform: rotate(180deg);
    transform: rotate(180deg);
  }
}

.emotion-0 {
  height: 35px;
  overflow: hidden;
  position: relative;
  width: 35px;
}

.emotion-0>div {
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  -webkit-animation: animation-0 0.6s 0s infinite linear;
  animation: animation-0 0.6s 0s infinite linear;
  background-image: linear-gradient(transparent 0%, transparent 70%, #ffffff 30%, #ffffff 100%);
  border-radius: 100%;
  border-width: 0px;
  height: 100%;
  position: absolute;
  width: 100%;
}

<div
  class="emotion-0"
>
  <div />
</div>
`;
